cli/src/semgrep/rule.py

```diff
@@ -249,6 +250,24 @@ def product(self) -> RuleProduct:
             else RuleProduct.sast
         )
 
+    @property
+    def scan_source(self) -> RuleScanSource:
```

Comment:
I think we do because we don't really have a reference to ```Rule``` object within the ```RuleMatch``` and only store the metadata. Also, there are places where we separately need to handle a set of ```Rule``` objects or ```RuleMatch``` objects and that's where this is super handy.